hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableTokenStore.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.node.app.service.mono.state.merkle.MerkleToken;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hedera.node.app.spi.state.WritableKVState;
+import com.hedera.node.app.spi.state.WritableKVStateBase;
+import com.hedera.node.app.spi.state.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Provides write methods for modifying underlying data storage mechanisms for
+ * working with Tokens.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail.
+ * This class is not complete, it will be extended with other methods like remove, update etc.,
+ */
+public class WritableTokenStore {
+    /** The underlying data storage class that holds the token data. */
+    private final WritableKVState<EntityNum, Token> tokenState;
+
+    /**
+     * Create a new {@link WritableTokenStore} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableTokenStore(@NonNull final WritableStates states) {
+        requireNonNull(states);
+
+        this.tokenState = states.get("TOKENS");
+    }
+
+    /**
+     * Persists a new {@link Token} into the state, as well as exporting its ID to the transaction
+     * receipt.
+     *
+     * @param token - the token to be mapped onto a new {@link MerkleToken} and persisted.
+     */
+    public void put(@NonNull final Token token) {
+        Objects.requireNonNull(tokenState).put(EntityNum.fromLong(token.tokenNumber()), Objects.requireNonNull(token));
```

Comment:
Objects.requireNonNull(tokenState) should be in the constructor

Reply:
Do we need requireNonNull if the first statement executed will result in an NPE? I think the end result is the same

Reply:
Yes ! Agreed. Changed. And removed `Objects.requireNonNull(tokenState)` checks. Thanks @hendrikebbers